### PR TITLE
Add EndpointSlice and new policy fields to syncer

### PR DESCRIPTION
**Pull request by caseydavenport**
## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
Support for Services in NetworkPolicy egress rules
```

### CODE REVIEW

File: lib/validator/v3/validator.go [hunk 1]
```diff
@@ -1203,6 +1226,22 @@ func validateNetworkPolicy(structLevel validator.StructLevel) {
 		}
 	}
 
+	// Services are only allowed on egress rules.
```
Comment (fasaxc) [inline lib/validator/v3/validator.go:1229]: Think we should also prevent services in the egress Destination struct

File: lib/validator/v3/validator_test.go [hunk 2]
```diff
@@ -1947,6 +1947,64 @@ func init() {
...
+									Name:      "service1",
+									Namespace: "default",
+								},
+							},
+						},
+					},
+				},
+			}, false,
+		),
+
```
Comment (fasaxc) [inline lib/validator/v3/validator_test.go:2007]: Worth shooting for full coverage in the validation logic; think there are a few more cases to cover

**Review by fasaxc — CHANGES_REQUESTED** [review]
Generally looking good to me

**Comment by caseydavenport** [comment]
@fasaxc I think this is ready for another look.

Also @mgleung this is the next part of the PR you reviewed today

File: lib/validator/v3/validator.go [hunk 3]
```diff
@@ -1333,6 +1387,33 @@ func validateGlobalNetworkPolicy(structLevel validator.StructLevel) {
...
```
Comment (mgleung) [inline lib/validator/v3/validator.go:1390]: Do we not also need to check that services are not set as the source on the egress rules for the GNP?

**Review by mgleung — APPROVED** [review]
LGTM

**Pull request merged**